x-pack/plugins/security/server/spaces/secure_spaces_client_wrapper.ts

```diff
@@ -108,62 +119,157 @@ export class SecureSpacesClientWrapper implements ISpacesClient {
       .filter(this.filterUnauthorizedSpaceResults);
 
     if (authorizedSpaces.length === 0) {
+      const error = Boom.forbidden();
+
       this.legacyAuditLogger.spacesAuthorizationFailure(username, 'getAll');
-      throw Boom.forbidden(); // Note: there is a catch for this in `SpacesSavedObjectsClient.find`; if we get rid of this error, remove that too
+      this.auditLogger.log(
+        spaceAuditEvent({
+          action: SpaceAuditAction.FIND,
+          error,
+        })
+      );
+
+      throw error; // Note: there is a catch for this in `SpacesSavedObjectsClient.find`; if we get rid of this error, remove that too
     }
 
     const authorizedSpaceIds = authorizedSpaces.map((space) => space.id);
+
     this.legacyAuditLogger.spacesAuthorizationSuccess(username, 'getAll', authorizedSpaceIds);
+    authorizedSpaces.forEach(({ id }) =>
+      this.auditLogger.log(
+        spaceAuditEvent({
+          action: SpaceAuditAction.FIND,
+          savedObject: { type: 'space', id },
+        })
+      )
+    );
 
     return authorizedSpaces;
   }
 
   public async get(id: string) {
     if (this.useRbac) {
-      await this.ensureAuthorizedAtSpace(
-        id,
-        this.authorization.actions.login,
-        'get',
-        `Unauthorized to get ${id} space`
-      );
+      try {
+        await this.ensureAuthorizedAtSpace(
+          id,
+          this.authorization.actions.login,
+          'get',
+          `Unauthorized to get ${id} space`
+        );
+      } catch (error) {
+        this.auditLogger.log(
+          spaceAuditEvent({
+            action: SpaceAuditAction.GET,
+            savedObject: { type: 'space', id },
+            error,
+          })
+        );
+        throw error;
+      }
     }
 
-    return this.spacesClient.get(id);
+    const space = this.spacesClient.get(id);
+
+    this.auditLogger.log(
+      spaceAuditEvent({
+        action: SpaceAuditAction.GET,
+        savedObject: { type: 'space', id },
+      })
+    );
```

Comment:
Calling this change out explicitly since it is different than the previous behaviour. 

I believe that we should log success scenarios here even if RBAC is disabled since we are now logging the fact that a CRUD operation took place rather than a pure authorisation check. Appreciate this scenario might never happen in real life (I am guessing the only way to disable RBAC is to disable security in which case audit logging would also be disabled) but the audit logger itself should be responsible for controlling when it logs events and when it doesn't. Authorisation failures obviously cannot happen when RBAC is disabled.